ember.md

```diff
@@ -0,0 +1,461 @@
+  lastName: DS.attr('string'),
+
+  fullName: Ember.computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+```
+
+### Don't use Ember's prototype extensions
+
+Avoid Ember's `Date`, `Function` and `String` prototype extensions. Prefer the
+corresponding functions from the `Ember` object.
+
+```javascript
+// Good
+
+const {
+  String: { w },
+  computed,
+  on,
+} = Ember;
+
+export default Model.extend({
+  hobbies: w('surfing skateboarding skydiving'),
+  fullName: computed('firstName', 'lastName', function _fullName() { ... }),
+  didError: on('error', function _didError() { ... })
+});
+
+// Bad
+
+export default Model.extend({
+  hobbies: 'surfing skateboarding skydiving'.w(),
+  fullName: function() { ... }.property('firstName', 'lastName'),
+  didError: function() { ... }.on('error')
+});
+```
+
+### Use `get` and `set`
+
+Calling `someObj.get('prop')` couples your code to the fact that
+`someObj` is an Ember Object. It prevents you from passing in a
+POJO, which is sometimes preferable in testing. It also yields a more
+informative error when called with `null` or `undefined`.
+
+Although when defining a method in a controller, component, etc. you
+can be fairly sure `this` is an Ember Object, for consistency with the
+above, we still use `get`/`set`.
+
+```js
+// Good
+import Ember from 'ember';
+
+const { get, set } = Ember;
+
+set(object, 'isSelected', true);
+get(object, 'isSelected');
+get(this, 'isPending');
+
+// Ok
+this.get('isPending');
+
+// Bad
+object.set('isSelected', true);
+object.get('isSelected');
+```
+
+### Use brace expansion
+
+This allows much less redundancy and is easier to read.
+
+Note that **the dependent keys must be together (without space)** for the brace expansion to work.
+
+```js
+// Good
+fullName: computed('user.{firstName,lastName}', function _fullName{
+  // Code
+}),
+
+// Bad
+fullName: computed('user.firstName', 'user.lastName', function _fullName{
+  // Code
+}),
+```
+
+### Trailing Commas
+
+Trailing commas must be used for multiline literals, and must not be used for single-line literals.
+
+```js
+// good
+const { get, computed } = Ember;
+const {
+  get,
+  computed,
+} = Ember;
+
+export default Ember.Object.extend({
+  foo() {
+  },
+
+  actions: {
+    bar() {
+    },
+  },
+});
+```
+
+*Trailing commas are easier to maintain; one can add or remove properties without having to maintain the commas. Also produces cleaner git diffs.*
+
+### All functions must be named
+
+Any function declared with the `function` keyword must have a name. Functions declared with the es6 short "fat arrow" syntax do not need a name.
+
+This helps with debugging, as the name of the function shows up in the backtrace instead of `(anonymous function)`.
+
+When the function is used as part of a macro to define a property, we have a convention of using the property name prefixed with an underscore.
+
+```js
+// good
+
+things.map((thing) => get(thing, 'name'));
```

Comment:
Does `things.map((thing) => get(thing, 'name'));` need to be here?  If yes, then I'd like to move `// Good` (capital "G") below it